src/modules/push-notifications/crons/push.notifications.energy.ts

```diff
@@ -0,0 +1,87 @@
+import { Injectable } from '@nestjs/common';
+import { Cron } from '@nestjs/schedule';
+import { Lock } from '@multiversx/sdk-nestjs-common';
+import { PushNotificationsService } from '../services/push.notifications.service';
+import {
+    AccountType,
+    NotificationType,
+} from '../models/push.notifications.types';
+import { ElasticAccountsEnergyService } from 'src/services/elastic-search/services/es.accounts.energy.service';
+import { ContextGetterService } from 'src/services/context/context.getter.service';
+
+@Injectable()
+export class PushNotificationsEnergyCron {
+    constructor(
+        private readonly pushNotificationsService: PushNotificationsService,
+        private readonly contextGetter: ContextGetterService,
+        private readonly accountsEnergyElasticService: ElasticAccountsEnergyService,
+    ) {}
+
+    @Cron('0 0 19 * * 5') // Every Friday at 19:00
```

Comment:
There is a better way to determine when the notifications should be sent. First of all we can keep this cron job to run every hour. We add a check at the beginning for the cron job:

`if((currentEpoch - feesCollector.firstWeekStartEpoch) % 7 !== 0) return;`

With this check in place we can use the same cron job to run on mainnet and devnet at every new start of the week.